_posts/2018-3-11-interactive-examples.md

```diff
@@ -0,0 +1,100 @@
+
+*"This is scoped to be a pretty small change."* - [me, January 2017](https://groups.google.com/d/msg/mozilla.dev.mdc/dhxwpNnlkSA/Vt7nnUFaEgAJ).
+
+Over the last year and a bit, the [MDN Web Docs] team has been designing, building, and implementing interactive examples for our reference pages. The motivation for this was the idea that MDN should do more to help "action-oriented" users: people who like to learn by seeing and playing around with example code, rather than by reading about it.
+
+We've just finished adding interactive examples for the JavaScript and CSS reference pages. This post looks back at the project to see how we got here and what we learned on the way.
+
+## First prototypes
+
+The project began as a suggestion by [Florian Scholz], at an MDN meetup in December 2016, that we should add simple examples to the tops of MDN reference pages. Initially we didn't plan to make them interactive: we thought that they would just be static code samples. It was I think [Julien Gattelier] and [Sebastian Zartner] who first suggested that they could be more interesting, with an editor that provided autocomplete suggestions. So I incorporated the wonderful [Code Mirror], and we built some prototypes to test.
+
+In the early prototypes the JS editor looked something like this:
+
+![](../images/interactive-examples/js-editor-first-user-test.png)
+
+The CSS editor looked something like this:
+
+![](../images/interactive-examples/css-editor-first-user-test.png)
+
+## First round of user testing
+
+In March 2017 [Kadir Topal] and I attended the first round of user testing, which was run by [Mark Hurst]. We learned a great deal about user testing, about our prototypes, and about what users wanted to see. We learned that users usually wanted to see examples and appreciated them being quick to find. Users liked interactive examples, too. But *noone* understood how to work our autocomplete, so reluctantly we let it go.
```

Comment:
Mark Hurst? This Mark Hurst? https://www.linkedin.com/in/markhurst1

Reply:
Yes! That Mark Hurst. He was lovely.